group_project/group_project.py

```diff
@@ -136,15 +136,31 @@ def milestone_dates(self):
         group_activity = GroupActivity.import_xml_string(self.data, self.is_admin_grader)
         return group_activity.milestone_dates
 
+    def _get_project_api_location(self):
+        location = '.project_api.ProjectAPI'
+        if hasattr(settings, 'GROUP_PROJECT_API_LOCATION'):
+            location = settings.GROUP_PROJECT_API_LOCATION
+
+        splitted = location.split('.')
+        return ".".join(splitted[:-1]), splitted[-1], not splitted[0]
+
     @property  # lazy
     def project_api(self):
         if self._project_api is None:
+            project_api_module, project_api_classname, relative = self._get_project_api_location()
+            if relative:
+                module = importlib.import_module(project_api_module, package='group_project')
+            else:
+                module = importlib.import_module(project_api_module)
```

Comment:
Can't you move the above logic inside `_get_project_api_location` and rename it to `_get_project_api()` ? That would be nicer.